kibana-reports/server/routes/utils/dataReportHelpers.ts

```diff
@@ -186,10 +186,10 @@ export const getEsData = (arrayHits, report, params) => {
 export const convertToCSV = async (dataset) => {
   let convertedData: any = [];
   const options = {
-    delimiter: { field: ',' },
+    delimiter: { field: ',', eol: '\n' },
     emptyFieldValue: ' ',
   };
-  await converter.json2csvAsync(dataset, options).then((csv) => {
+  await converter.json2csvAsync(dataset[0], options).then((csv) => {
```

Comment:
Thanks for the fix. Do you happen to know what caused this? Earlier I tested by UT and manual IT but didn't see this issue.

If necessary, please change mock data accordingly in https://github.com/opendistro-for-elasticsearch/kibana-reports/blob/dev/kibana-reports/server/routes/utils/__tests__/savedSearchReportHelper.test.ts#L371.

Reply:
This line changes the format of the data string to be parsed correctly into the CSV file on the client-side. The issue was in the `sanitize()` method in `dataReportHelpers`, it threw the error when a non-string field tried to be parsed with `startsWith` (e.g AvgTicketPrice: 143.83 or something similar).

Reply:
I see. You're talking about the changes in `sanitize()` right? Curious why `dataset` is changed to `dataset[0]`? Does the UT still pass?

Reply:
Yeah, talking about the changes in sanitize(). Changing `dataset` to `dataset[0]` allowed the data string to be parsed into the CSV file correctly when exporting- need to modify the UTs